File: qtech/inspection/wb-olp-inspection/src/main/java/com/im/inspection/config/DppConfigManager.java
package com.im.inspection.config;

import org.im.config.ConfigurationManager;
import org.im.config.impl.DefaultConfigurationManager;

/**
 * 应用统一配置管理器
 *
 * @author gaozhilin
 * @email gaoolin@gmail.com
 * @date 2025/10/16
 */

public class DppConfigManager {
    private static final Object lock = new Object();
    private static volatile ConfigurationManager instance;

    public static ConfigurationManager getInstance() {
        if (instance == null) {
            synchronized (lock) {
                if (instance == null) {
                    instance = new DefaultConfigurationManager();
                }
            }
        }
        return instance;
    }
}

File: qtech/datadev/wb-wire-usage-etl/src/main/scala/com/im/qtech/datadev/wb/usage/config/DppConfigManager.scala
package com.im.qtech.datadev.wb.usage.config

import org.im.config.ConfigurationManager
import org.im.config.impl.DefaultConfigurationManager

/**
 * 应用统一配置管理器
 *
 * @author gaozhilin
 * @email gaoolin@gmail.com
 * @date 2025/10/16
 */
object DppConfigManager {
  private lazy val instance: ConfigurationManager = new DefaultConfigurationManager()

  def getInstance(): ConfigurationManager = instance
}
